build-zip.sh

```diff
@@ -38,62 +31,65 @@ function get_repo_base_url() {
   esac
 
   echo $BASE_URL
-} && export -f get_repo_base_url
+}
 
 function download_repo_index() {
-  local DL_URL="$(get_repo_base_url $1)/index.xml"
+  local DL_URL="$(get_repo_base_url "$1")/index.xml"
   local FILE_NAME="$1_index.xml"
 
-  fetch $DL_URL $FILE_NAME
-} && export -f download_repo_index
+  fetch "$DL_URL" "$FILE_NAME"
+}
 
 function xpath_exec() {
-  local INDEX_FILE=$1
+  local INDEX_FILE="$1"
   local XPATH_CMD="$2"
 
-  xmlstarlet select -t -v "$XPATH_CMD" $INDEX_FILE
-} && export -f xpath_exec
+  xmlstarlet select -t -v "$XPATH_CMD" "$INDEX_FILE" | head -1
```

Comment:
If the output of the command is bigger than a line, one should fix the XPath query, not the command.

Reply:
This may need to assume that the configuration file is correct

Reply:
If the config file is wrong, the script should fail. Otherwise is bad practice.